### DISCUSSION TITLE
Left nav menu 404 after publish to IIS application dir instead of root

### DISCUSSION THREAD

**Discussion by fdgit** [serenity-is/Serenity#5585] [Help]
Are there changes need to be made to publish to an application subdirectory 
in IIS? (for example http://myapp/serene as oppesed to  http://serene).

When I publish to an application directory ("/myapp") rather than the root, the login page works 
but my left menu items are all missing /myApp/, so the links are 404.

If I add a Url parameter to the  assembly: NavigationLink(..., Url="myapp/myThing") in *navigation.cs, then the 
left menu URLs are correct (they begin with /myApp/), but the drilldown into their dialogs fails to find the dialog templates
and the urls appear malformed (like this: http://localhost/myapp/#categorypart-subItempart/12345)

In desperation have tried 
- adding app.UsePathBase("/myApp/"); to startup cs
- adding a <base href="/myapp"> tag
- adding ~ in the Route on the controller action.

I wrote a small core core app and all its links are correct when published to /myapp/ rather than root, and
I see the demo is hosted under an application directory, so I must be missing a key point)
Thank you

**Comment by VictorTomaili** [comment]
It's not serenity related. It's just configuration a .net 5 application for IIS. Because of that I convert your issue to discussion. @fdgit Please use discussion for Q&A next time.

**Comment by fdgit** [comment]
I had to add the correct URL parameter to the NavigationLink attributes, being sure that they matched each modules *Form.cs:

-[assembly: NavigationLink(9000, "Administration/User Management", typeof(Administration.UserController), icon: "fa-users")]
+[assembly: NavigationLink(9000, "Administration/User Management", typeof(Administration.UserController), icon: "fa-users", **Url = "Administration/User"**)]
\
